include/vast/Dialect/HighLevel/HighLevelUnsupported.td

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022-present, Trail of Bits, Inc.
+
+#ifndef VAST_DIALECT_HIGHLEVEL_IR_HIGHLEVELUNSUPPORTED
+#define VAST_DIALECT_HIGHLEVEL_IR_HIGHLEVELUNSUPPORTED
+
+def HighLevel_UnsupportedDecl
```

Comment:
We want to have a separate dialect for unsupported operations.

Reply:
@lkorenc, Should the unsupported operation dialect named differently. I used `HighLevel_` as prefix because it is high level dialect. The dialects are of three types. The `UnsupportedDecl` is used only for lowering the Decl node. `UnsupportedExpr` and `UnsupportedOp` are used for lowering stmt node.

Could you explain what do you mean by saying "want to have a separate dialect for unsupported operations"?

Reply:
Yeah I do not want it to be part of high level dialect. I would like a separate dialect - unsupported - same way as we have `ll` and `core` etc. alongside the `hl`. And this unsupported dialect will have those 3 operations.

Or do you think there is a reason this cannot be achieved/is not a good idea?

Reply:
@lkorenc, Thanks for explaining it. We can have `unsupported` as a different set of dialect similar to `core`, or `low-level`. The one concern I have is the coverage scope for these dialects are limited. In that case is it a good idea to have them as separate dialect set? I am not sure.

I will have these changes in a separate commit and we can review how much complexity does it add, if we introduce a new set of dialect for unsupported. We can then decide if we should go ahead with that or not.

Reply:
I think we can remove this now. Unsupported dialect is the way to go, because we can have eventually something that does not use `hl` dialect but still leverage `unsupported` stuff.